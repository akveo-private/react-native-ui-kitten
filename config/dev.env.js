const path = require('path');

const evaPath = path.resolve(__dirname, '../../eva');

module.exports = {
  ENV: 'dev',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(evaPath, 'packages/mapping/eva'),
  THEME_PATH: path.resolve(evaPath, 'packages/theme/eva'),
};
