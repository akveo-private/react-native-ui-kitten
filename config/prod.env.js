const path = require('path');

const playgroundPath = path.resolve(__dirname, '../src/playground');

module.exports = {
  ENV: 'prod',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(playgroundPath, 'node_modules/@eva/eva'),
  THEME_PATH: path.resolve(playgroundPath, 'node_modules/@eva/theme-eva'),
};
