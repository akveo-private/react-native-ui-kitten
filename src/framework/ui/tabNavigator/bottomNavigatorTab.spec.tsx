import React from 'react';
import { render } from 'react-native-testing-library';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
} from '@kitten/theme';
import * as config from './bottomNavigatorTab.spec.config';
import {
  BottomNavigatorTab as BottomNavigatorTabComponent,
  Props as BottomNavigatorTabProps,
} from './bottomNavigatorTab.component';

const BottomNavigatorTab = styled<BottomNavigatorTabComponent, BottomNavigatorTabProps>(BottomNavigatorTabComponent);

describe('@bottom-navigator-tab: component checks', () => {

  const testTabUri: string = 'https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/examples.png';

  const Mock = (props?: BottomNavigatorTabProps): React.ReactElement<StyleProviderProps> => (
    <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
      <BottomNavigatorTab {...props}/>
    </StyleProvider>
  );

  it('* empty', () => {
    const component = render(
      <Mock/>,
    );
    expect(component).toMatchSnapshot();
  });

  it('* with icon', () => {
    const getImageSource = () => ({ uri: testTabUri });
    const component = render(
      <Mock getIconSource={getImageSource}/>,
    );
    expect(component).toMatchSnapshot();
  });

  it('* text/selected', () => {
    const getImageSource = () => ({ uri: testTabUri });
    const component = render(
      <Mock
        getIconSource={getImageSource}
        selected={true}
        title='Test'
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('* text/unselected', () => {
    const getImageSource = () => ({ uri: testTabUri });
    const component = render(
      <Mock
        getIconSource={getImageSource}
        selected={false}
        title='Test'
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('* getIconSource', () => {
    const getImageSource = jest.fn();
    const component = render(<Mock getIconSource={getImageSource}/>);
    expect(getImageSource).toHaveBeenCalled();
  });

});
