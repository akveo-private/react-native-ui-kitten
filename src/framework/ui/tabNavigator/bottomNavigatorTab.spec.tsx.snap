// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@bottom-navigator-tab: component checks * empty 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
/>
`;

exports[`@bottom-navigator-tab: component checks * text/selected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/examples.png",
      }
    }
    style={
      Object {
        "height": 40,
        "marginBottom": undefined,
        "tintColor": "#3366FF",
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#3366FF",
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`@bottom-navigator-tab: component checks * text/unselected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/examples.png",
      }
    }
    style={
      Object {
        "height": 40,
        "marginBottom": undefined,
        "tintColor": "#8992A3",
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#8992A3",
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`@bottom-navigator-tab: component checks * with icon 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/examples.png",
      }
    }
    style={
      Object {
        "height": 40,
        "marginBottom": undefined,
        "tintColor": "#8992A3",
        "width": 40,
      }
    }
  />
</View>
`;
